examples/lazy_read_all_values.rs

```diff
@@ -41,7 +41,7 @@ mod lazy_reader_example {
         // We're going to recursively visit and read every value in the input stream, counting
         // them as we go.
         let mut count = 0;
-        let mut reader = LazyReader::new(ion_data)?;
+        let mut reader = LazyBinaryReader::new(ion_data)?;
```

Comment:
Is the user required to know whether the data is binary or text ahead of time? (Or this just a temporary thing because the lazy text reader is not implemented yet?)